pkg/triggers/service.go

```diff
@@ -80,7 +84,16 @@ func (svc *service) Run(triggerName string, vars map[string]interface{}) ([]Cond
 		}
 
 		if condition.OncePer != "" {
-			if oncePer, ok, err := unstructured.NestedFieldNoCopy(vars, strings.Split(condition.OncePer, ".")...); err == nil && ok {
```

Comment:
Instead of using regex can we use `https://github.com/antonmedv/expr` , similar to:

https://github.com/argoproj-labs/argocd-notifications/blob/4673856e18e8d34cb60cd0d7515f12030f53197e/pkg/triggers/service.go#L76

Simple `oncePer` expression (e.g. `app.status.operationState.phase`) should keep working as is. For complex cases user would be able to write e.g `app.status.operationState.phase + '/commit'` . I think it is better to use `https://github.com/antonmedv/expr` since we already using it for trigger conditions and it provides more features.

Reply:
Thank you for reviewing. I think so too that using `https://github.com/antonmedv/expr` is better than regex.
I'll fix it.